src/AudioUnitSDK/AUInputElement.cpp

```diff
@@ -8,8 +8,8 @@ namespace ausdk {
 
 constexpr bool HasGoodBufferPointers(const AudioBufferList& abl, UInt32 nBytes) noexcept
 {
-	const AudioBuffer* buf = abl.mBuffers;                // NOLINT
-	for (UInt32 i = abl.mNumberBuffers; i-- > 0; ++buf) { // NOLINT
+	const AudioBuffer* buf = abl.mBuffers;                   // NOLINT
+	for (UInt32 i = abl.mNumberBuffers; i > 0; i--, ++buf) { // NOLINT
```

Comment:
why?

Reply:
Post decrement is better in this case. Because unless I is used, it is best to not deal with unneeded underflows

Reply:
`i` will not underflow since the loop exits upon `i == 0` and before what would be the following decrement (or fails to ever iterate if `i` is initialized to zero.